test/regression/regression_test.go

```diff
@@ -0,0 +1,382 @@
+	daemons := make([]*core.DrandDaemon, n)
+	for i := 0; i < newN; i++ {
+		// we have to self sign the keys, as the CLI normally does this for us
+		_, _, err := core.SelfSignKeys(log.DefaultLogger(), fmt.Sprintf("%s/multibeacon", nodes[i].dir))
+		require.NoError(t, err)
+
+		opts := []core.ConfigOption{
+			core.WithConfigFolder(nodes[i].dir),
+			core.WithPrivateListenAddress(nodes[i].private),
+			core.WithPublicListenAddress(nodes[i].public),
+			core.WithControlPort(nodes[i].control),
+			core.WithDBStorageEngine(chain.BoltDB),
+		}
+		conf := core.NewConfig(log.DefaultLogger(), opts...)
+		d, err := core.NewDrandDaemon(context.Background(), conf)
+		require.NoError(t, err)
+		require.NoError(t, d.LoadBeaconsFromDisk(context.Background(), "", false, "default"))
+
+		daemons[i] = d
+	}
+
+	// wait for all the nodes to start
+	for i := 0; i < newN; i++ {
+		require.NoError(t, waitForNodeStarted(nodes[i].public, 20))
+	}
+
+	// we pull the list of participants from last epoch
+	status, err := daemons[0].DKGStatus(context.Background(), &dkgproto.DKGStatusRequest{BeaconID: "default"})
+	require.NoError(t, err)
+	// and get the leaver - their key won't be available to create the next proposal
+	leavingNode := nodes[n-1]
+	var leaver *dkgproto.Participant
+	for _, participant := range status.Complete.Joining {
+		// the addresses start from 0, so n-1
+
+		if participant.Address == leavingNode.private {
+			leaver = participant
+		}
+	}
+
+	// now that the daemons are started, we extract their identities and set up some plumbing for the DKG
+	runners := make([]*dkg.TestRunner, newN)
+	remainers := make([]*dkgproto.Participant, newN)
+	for i := 0; i < newN; i++ {
+		client, err := net.NewDKGControlClient(log.DefaultLogger(), nodes[i].control)
+		require.NoError(t, err)
+		runners[i] = &dkg.TestRunner{
+			Client:   client,
+			BeaconID: "default",
+			Clock:    clockwork.NewRealClock(),
+		}
+
+		ident, err := daemons[i].GetIdentity(context.Background(), &drand.IdentityRequest{Metadata: &drand.Metadata{BeaconID: "default"}})
+		require.NoError(t, err)
+		remainers[i] = &dkgproto.Participant{
+			Address:   ident.Address,
+			Key:       ident.Key,
+			Signature: ident.Signature,
+		}
+	}
+
+	// then we actually do the reshare without node 3
+	require.NoError(t, runners[0].StartReshare(2, 1, nil, remainers, []*dkgproto.Participant{leaver}))
+	require.NoError(t, runners[1].Accept())
+	require.NoError(t, runners[0].StartExecution())
+	require.NoError(t, runners[0].WaitForDKG(log.DefaultLogger(), 2, 60))
+}
+
+func runCommand(cmd string, args ...string) error {
+	c := exec.Command(cmd, args...)
+	c.Stdout = os.Stdout
+	c.Stderr = os.Stderr
+	return c.Run()
+}
+
+func waitForNodeStarted(publicAddress string, seconds int) error {
+	var err error
+	remaining := seconds
+	for remaining > 0 {
+		res, err := http.Get(fmt.Sprintf("http://%s/health", publicAddress))
+		if err == nil || (res != nil && res.StatusCode != 200) {
```

Comment:
That's because a non-2xx response doesn't cause an error with `http.Get`, right?

Reply:
yup